draft-jholland-quic-multicast.md

```diff
@@ -186,12 +186,15 @@ The different path can be either the unicast connection or another multicast cha
 
 Stream IDs in channels are restricted to unidirectional server initiated streams, or those with the least significant 2 bits of the stream ID equal to 3 (see {{RFC9000}} Section 2.1).
 
-Since a server has access to all data in channels it uses, a server can always avoid stream ID collisions with the stream IDs carried in channels, and can usually (depending on the timing) avoid allowing channels to exceed the client's max_streams_uni by requesting that clients leave channels before their limits would be exceeded.
+When a channel contains streams with ids above the client's unidirectional MAX_STREAMS, the server MUST NOT instruct the client to join that channel and SHOULD send a STREAMS_BLOCKED frame, as described in Sections 4.6 and 19.14 of {{RFC9000}}.
 
-However, since clients can join later than a channel began, clients supporting the multicast extensions to QUIC should be prepared to handle stream IDs that do not begin at early values, since by the time a client joins a channel in progress the stream id count might have been increasing for a long time.
-Clients should therefore begin with a high initial_max_streams_uni or send an early MAX_STREAMS type 0x13 value (see Section 19.11 of {{RFC9000}}) with a high limit.
+If the client is already joined to a channel that carries streams that exceed or will soon exceed the client's unidirectional MAX_STREAMS, the server MUST send a MC_CHANNEL_LEAVE frame.
+
+If a client receives a STREAM frame with an ID above its MAX_STREAMS on a channel, the client MAY increase its unidirectional MAX_STREAMS to a value greater than the new ID and send an update to the server, otherwise it MUST drop the packet and leave the channel with reason Max Streams Exceeded.
```

Comment:
Wouldn‘t it be enough to just have the client react to the streams exceeding the limit? I think having the server also send a LEAVE frame might not always be perfect, especially in cases where the client just can increase its stream limit. Also, LEAVE frames don’t contain a reason for leaving, which means the client wouldn’t necessarily know why it has to leave the channel. So far it’s mostly clients leaving channels and stating the reason (e.g. rate exceeded).


Also, CHANNEL_STATE Frames Are missing the new reason now I think.

Reply:
That can be a SHOULD instead of a MUST.  I feel like the server should be proactively responsible here if client is not moving its MAX_STREAMS, but it doesn't have to be a hard limit, especially since the server can be out of sync with the channel sender by a bit anyway.

Hmm, I thought I added the reason, but maybe it's in a different branch.  I may have confused myself...